state_machine/applications/system-check/tests/nvm_access_test.py

```diff
@@ -0,0 +1,139 @@
+    Test to make sure all nvm counters and flags exist, and that their values are in range.
+    Also check that multibit flags do not interfere with each other or reference the same
+    region in non volatile memory.
+    """
+
+    print("Starting NVM Test...")
+    nvm_counters_exist = [(counter is not None) for counter in nvm_counters_values]
+    nvm_flags_exist = [(flag is not None) for flag in nvm_flags_values]
+
+    counter_flag_access = not (False in nvm_counters_exist) and not (False in nvm_flags_exist)
+    counter_flag_access_string = ""
+    if counter_flag_access:
+        counter_flag_access_string = "All counters and flags are accessible."
+    else:
+        counter_flag_access_string = "The following counters / flags are None:"
+    print(nvm_counters_items, nvm_flags_items)
+
+    nvm_counters_inrange = [verify_counter_bits(counterstr, counter) for counterstr, counter in nvm_counters_items]
+    nvm_flags_inrange = [verify_flag_bits(flagstr, flag) for flagstr, flag in nvm_flags_items]
+
+    counter_flag_inrange = (not (False in nvm_counters_inrange)) and (not (False in nvm_flags_inrange))
+    counter_flag_inrange_string = ""
+    if counter_flag_inrange:
+        counter_flag_inrange_string = "All existing counters and flags are in range."
+    else:
+        counter_flag_inrange_string = "The following counters / flags are not in range:"
+
+    for i in range(len(nvm_counters_values)):
+        # if all counters are accessible, we'll never reach this
+        # we only add to counter_flag_access_string if something is inaccessible
+        if not nvm_counters_exist[i]:
+            counter_flag_access_string += nvm_counters_items[i][0] + "; "
+        # if the counter exists, verify its values
+        else:
+            # if all counters are in range, we'll never reach this
+            # we only add to counter_flag_inrange_string if something is out of range
+            if not nvm_counters_inrange[i]:
+                counter_flag_inrange_string += nvm_counters_items[i][0] + "; "
+
+    for i in range(len(nvm_flags_values)):
+        # if all flags are accessible, we'll never reach this
+        # we only add to counter_flag_access_string if something is inaccessible
+        if not nvm_flags_exist[i]:
+            counter_flag_access_string += nvm_flags_items[i][0] + "; "
+        # if the flag exists, verify its values
+        else:
+            # if all flags are in range, we'll never reach this
+            # we only add to counter_flag_inrange_string if something is out of range
+            if not nvm_flags_inrange[i]:
+                print(nvm_flags_inrange[i])
+                counter_flag_inrange_string += nvm_flags_items[i][0] + "; "
+
+    result_dict["NVM_CounterFlagAccess"] = (counter_flag_access_string, counter_flag_access)
+    result_dict["NVM_CounterFlagValuesInRange"] = (counter_flag_inrange_string, counter_flag_inrange)
+    counter_overflow, counter_overflow_string = check_nvm_counter_overflow()
```

Comment:
It would be nice to group the logic for each test, instead of having it more spread out throughout the function